internal/services/appservice/static_web_app_link_backend.go

```diff
@@ -0,0 +1,241 @@
+					loc = location.Normalize(appModel.Location)
+				}
+
+			} else if containerAppId, err := containerapps.ParseContainerAppID(model.BackendResourceId); err == nil {
+				linkedBackend.id = containerAppId.ID()
+				linkedBackend.name = containerAppId.ContainerAppName
+
+				backendClient := metadata.Client.ContainerApps.ContainerAppClient
+
+				containerApp, err := backendClient.Get(ctx, *containerAppId)
+				if err != nil {
+					return fmt.Errorf("reading specified %s: %+v", *containerAppId, err)
+				}
+
+				if appModel := containerApp.Model; appModel != nil {
+					loc = location.Normalize(appModel.Location)
+				}
+
+			} else {
+				return fmt.Errorf("unsupported backend resource type")
+			}
+
+			id := staticsites.NewLinkedBackendID(staticAppId.SubscriptionId, staticAppId.ResourceGroupName, staticAppId.StaticSiteName, linkedBackend.name)
+
+			existing, err := client.GetLinkedBackend(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			backends, err := client.GetLinkedBackends(ctx, *staticAppId)
+			if err != nil {
+				return fmt.Errorf("checking for existing Static Site backends for %s: %+v", id, err)
+			}
+
+			if backendList := backends.Model; backendList != nil {
+				if len(*backendList) != 0 {
+					return fmt.Errorf("%s already has a backend and cannot have another", id)
+				}
+			}
+
+			payload := staticsites.StaticSiteLinkedBackendARMResource{
+				Properties: &staticsites.StaticSiteLinkedBackendARMResourceProperties{
+					Region:            pointer.To(loc),
+					BackendResourceId: pointer.To(linkedBackend.id),
+				},
+			}
+
+			if err = client.LinkBackendThenPoll(ctx, id, payload); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r StaticWebAppLinkBackendResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AppService.StaticSitesClient
+			state := StaticWebAppLinkBackendModel{}
+
+			id, err := staticsites.ParseLinkedBackendID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			state.StaticWebAppID = staticsites.NewStaticSiteID(id.SubscriptionId, id.ResourceGroupName, id.StaticSiteName).ID()
+
+			result, err := client.GetLinkedBackend(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(result.HttpResponse) {
+					return metadata.MarkAsGone(*id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if model := result.Model; model != nil {
+				if props := model.Properties; props != nil {
+					state.BackendResourceId = pointer.From(props.BackendResourceId)
```

Comment:
We'll need to check the ID type here and parse it with the appropriate parser to ensure the ID is correctly cased and formatted for the state.